data/src/main/java/org/cryptomator/data/cloud/crypto/CryptoImplVaultFormatPre7.kt

```diff
@@ -128,6 +128,18 @@ internal class CryptoImplVaultFormatPre7(
 			.filterIsInstance<CloudFile>()
 			.map { node ->
 				ciphertextToCleartextNode(cryptoFolder, dirId, node)
+			}.onEach { cryptoNode ->
+				if (cryptoNode is CryptoFile && isImageMediaType(cryptoNode.name)) {
+					val cacheKey = generateCacheKey(cryptoNode.cloudFile)
+					cryptoNode.cloudFile.cloud?.type()?.let { cloudType ->
+						getLruCacheFor(cloudType)?.let { diskCache ->
+							val cacheFile = diskCache[cacheKey]
+							if (cacheFile != null) {
+								cryptoNode.thumbnail = cacheFile
+							}
+						}
+					}
+				}
```

Comment:
The same as in the other list implementation, did you check the performance impact?